clash-prelude/src/Clash/Sized/Internal/Unsigned.hs

```diff
@@ -498,18 +496,21 @@ shiftL#, shiftR#, rotateL#, rotateR# :: forall n .KnownNat n => Unsigned n -> In
 shiftL# =
   \(U v) i ->
     if i >= 0 then
-#if MIN_VERSION_base
-      U ((naturalShiftL v i) `mod` m)
+#if MIN_VERSION_base(4,15,0)
+      let i' = fromIntegral i
+       in U ((naturalShiftL v i') `mod` m)
 #else
       U ((shiftL v i) `mod` m)
 #endif
     else
       error ("'shiftL undefined for negative number: " ++ show i)
  where
 #if MIN_VERSION_base(4,15,0)
-  m = 1 `naturalShiftL` naturalToWord (natVal (Proxy @n))
+  sz = naturalToWord (natVal (Proxy @n))
```

Comment:
`natToNum @n`?

Reply:
Eh, my feeling is that these internal modules could do with a little spring cleaning in places anyway. Leave it for another PR?

Reply:
Sure thing. I commented because you changed it, that's all :)

Reply:
If this is approved, feel free to merge :wink:

Reply:
Please check that the generated Core is as efficient if you do decide to switch to `natToNum`.